tests/datasets/test_geo.py

```diff
@@ -212,6 +221,29 @@ def sentinel(self, request: SubRequest) -> Sentinel2:
     def test_files(self, paths: Union[str, Iterable[str]]) -> None:
         assert 1 <= len(NAIP(paths).files) <= 2
 
+    def test_files_property_warns_non_existing_file(self, tmp_path: Path) -> None:
```

Comment:
Since we're testing `GeoDataset.files`, can we move these tests to `TestGeoDataset`? We can also rename `MockRasterDataset` to `MockGeoDataset`.

Reply:
Totally makes sense. I pushed a suggestion. GeoDataset does not accept arg `paths`, so I added it to the existing `CustomGeoDataset` in the test. Removed the MockRasterDataset as GeoDataset.__init__ does not try to open the paths.